R/xgb_binning.R

```diff
@@ -0,0 +1,250 @@
+#' Discretize numeric variables with XgBoost
+#'
+#' `step_xgb_binning` creates a *specification* of a recipe
```

Comment:
Can we call it `step_discretize_tree`? It would keep it more consistent with the existing one and avoid being method-specific in case we want a different underlying model at some later date.

Reply:
Makes sense - changing